bzt/resources/selenium_taurus_extras.py

```diff
@@ -40,3 +41,42 @@ def apply(self, template):
         self.tmpl.template = template
         self.tmpl.variables = self.variables
         return str(self.tmpl)
+
+
+class WindowManager:
+
+    def __init__(self, driver):
+        self.driver = driver
+        self.windows = {}
+
+    def switch(self, window_name=None):
```

Comment:
That's a bit of a complex method with a lot of nested conditionals. Can you split it into multiple ones (like `switch_by_index`, `switch_by_seqname`, `switch_by_name` or something like that) ?

Reply:
Hi @dmand 
Ok, I reduce the cyclomatic complexity to -4
587c1f55bf6a04234b7ce29eb749c9ebc8e1b8c2

My commit reviewed
https://app.codacy.com/app/3dgiordano/taurus_abstracta/commit?bid=7267532&cid=200105074